### PR TITLE
runtime: Don't hardcode PATH_MAX

### PR DESCRIPTION
**Pull request by Nopey**
Windows' _fullpath and _getcwd will handle allocation if we pass NULL as the buffer, but please note that this [will not remove path length limitations](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation).

Unfortunately, posix_fs.jakt is not so straightforward.
realpath: POSIX.1-2008 lets us pass NULL for the second argument, resolved_path; libc will allocate for us.
getcwd: Linux lets us pass NULL, but POSIX defines this as undefined behavior. Can and should we special-case any os/libc whose getcwd will do the allocation for us?

### CODE REVIEW

**Comment by alimpfard** [comment]
We could add the means to select different modules based on libc, but iirc getcwd() will return with an error if the path doesn't fit in the buffer, so we should be able to keep allocating larger buffers until it returns non-NULL?

**Comment by Nopey** [comment]
Paths over 4096 can now be returned by jakt's current_directory and real_path: We use the POSIX.1-2008 realpath behavior, and getcwd has a loop to grow the buffer size until the buffer is sufficient.

This does **not** mean Jakt is long path aware on POSIX, as that would require breaking paths up into relative paths that fit within PATH_MAX ([which could be as low as _POSIX_PATH_MAX, 256](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html)) and using openat chaining (to implement filesystem apis that may fail on long paths such as as [mkdir](https://pubs.opengroup.org/onlinepubs/009695299/functions/mkdir.html))

**Review by alimpfard — COMMENTED** [review]
Apparently I forgot to press "comment" the last time I came to review :sweat_smile:

File: runtime/jaktlib/platform/posix_errno.jakt [hunk 1]
```diff
@@ -7,3 +7,9 @@ fn errno_value() -> i32 {
         return *__errno_location()
     }
 }
+
+fn erange_value() -> i32 {
```
[the remark this answers is not in the document]
Comment (Nopey) [reply runtime/jaktlib/platform/posix_errno.jakt:12]: Added a comment to this effect
```c
// FIXME: hacky helper until Jakt supports importing C macros
```

**Review by alimpfard — APPROVED** [review]

**Pull request merged**